### ISSUE TITLE
unused-import warning when using fully qualified java references

### ISSUE THREAD

**Issue by ikappaki** [clj-kondo/clj-kondo#1933]
**version**

clj-kondo v2022.12.10

**platform**

native MS-Windows

**editor**

command line

**problem**

`clj-kondo` gives a warnings about java import being used, although it is referenced.

**repro**

1. Create an `issue.clj` file with a java import that is being referenced using the FQD name
``` clojure
(ns issue
  (:import java.util.Date))

(java.util.Date.)
```
2. Run `clj-kondo --lint issue.clj`, you should get a warning
```
issue.clj:2:23: warning: Unused import Date
linting took 4ms, errors: 0, warnings: 1
```
**expected behavior**

No warning should be produced since the import is used.

There is no warning if just the unqualified reference is used, e.g.

```clojure
(Date.)
```

Thanks

**Comment by borkdude** [comment]
> There is no warning if just the unqualified reference is used, e.g.

This is exactly the point. The import introduces `Date` as an unqualified reference into your namespace. And this import is not used when you write `(java.util.Date.)`. An import doesn't load JVM classes, you can write `(java.util.Date.)` also without eh `:import`. An import is only used to bring unqualified names into the scope of your namespace.

**Comment by ikappaki** [comment]
Thanks, I didn't realize that import was only used to introduce a mapping 😮
